### ISSUE TITLE
WebAssembly 2022

### ISSUE THREAD

**Issue by rviscomi** [HTTPArchive/almanac.httparchive.org#2885]
# WebAssembly 2022

![WebAssembly illustration](https://raw.githubusercontent.com/HTTPArchive/almanac.httparchive.org/main/src/static/images/2021/webassembly/hero_lg.jpg)

If you're interested in contributing to the WebAssembly chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @ColinEberhardt | @ColinEberhardt | @binji, @RReverser  | @JamieWhitMac | @rviscomi  | @siakaramalegos  |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [x] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [x] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 WebAssembly resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-webassembly on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/1U8CeQ_2ga_m-TGDAIGgB2PjB0O6Z5dauqtvuZubmXtQ/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/webassembly
[~google-sheets]: https://docs.google.com/spreadsheets/d/11jyABro0fKtuN6INnYP9pJlv5QWwp0jfJyTsGfKgScg/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/webassembly.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

**Comment by ColinEberhardt** [comment]
Hi, I’m Colin Eberhardt, the author of the WebAssembly Weekly newsletter and general wasm enthusiast 👋 

I’m interested in the author / analyst roles, but would be good to discuss the detail, make sure I am comfortable with the commitment.

**Comment by siakaramalegos** [comment]
Hi, @ColinEberhardt that's great news! You can find out more about each role in the wiki - see the side bar for each one: https://github.com/HTTPArchive/almanac.httparchive.org/wiki

The author is usually the content lead as well since it makes sense to guide the team where you want the writing to go. For the analyst role, anyone who knows SQL decently could work. Last year I did most of the analyst work for the chapter I authored, so it's definitely doable to do both, though more work. Rick is going to lead a workshop for analysts sometime this week so new people are welcome! If you know other experts in the space, it would be great to get them to sign up as reviewers. Having multiple reviewers is better than just one, if possible.

**Comment by siakaramalegos** [comment]
@ColinEberhardt here's the workshop for anyone interested in being an analyst! https://twitter.com/HTTPArchive/status/1516060214109298693

**Comment by siakaramalegos** [comment]
@ColinEberhardt I sent you an email to do a check-in - I know we're still waiting for some more volunteers but want to get you started so that doesn't hold you back. Let me know if you didn't get it.

**Comment by ColinEberhardt** [comment]
Hu @siakaramalegos - I’ve not spotted that email. Was it to my gmail address? F can you resend, or perhaps DM via Twitter?

**Comment by siakaramalegos** [comment]
@ColinEberhardt just re-sent it to the gmail address in your GH profile. It doesn't look like I can DM you on Twitter, but feel free to DM me if you don't see the email!

**Comment by siakaramalegos** [comment]
@ColinEberhardt we need some more volunteers to make this chapter viable and achieve the first milestone on May 1. It puts the chapter at risk of being dropped if we can't make a viable team. Can you recruit some more people? For viability we need at least 1 reviewer and 1 analyst in addition to at least 1 author. It's nice to have 3+ reviewers though.

**Comment by binji** [comment]
I'd like to help out here too! I think I'd do best as a reviewer, but perhaps could help out in other aspects as well.

**Comment by rviscomi** [comment]
@ColinEberhardt @binji to confirm, are either of you interested in taking on the [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide) role?

**Comment by ColinEberhardt** [comment]
That's great @binji - I'll be in touch shortly, it would be great to chat.

@rviscomi I could potentially take on the analyst role, but am looking to 'recruit' one of my colleagues. I should know in the next day or so.

**Comment by rviscomi** [comment]
@ColinEberhardt @binji I know we're still figuring out the structure of the team, but it'd be great to make sure you have access to the [planning doc](https://docs.google.com/document/d/1U8CeQ_2ga_m-TGDAIGgB2PjB0O6Z5dauqtvuZubmXtQ/edit?usp=sharing) and start adding your ideas to the outline. What's new in WASM this year, or what would be interesting to revisit from last year? We've [added the ability](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2914#issuecomment-1103060612) for the WebPageTest agents to analyze WASM files at runtime, so if you need to make any changes to it (new metrics or any other improvements), those need to be added before the June crawl kicks off. So we're hoping to have the outline completed on May 15, to give us an idea of what changes need to be made and leave enough time to get those changes in. Thanks!

**Comment by JamieWhitMac** [comment]
Hi all, after chatting with @ColinEberhardt I've agreed to take on the analyst role for this chapter.

Looking forward to working on this, finding out what's new with WASM usage, and how it's changing the web.

**Comment by siakaramalegos** [comment]
@ColinEberhardt go ahead and update the table and check off milestone 1 - congrats!!

**Comment by ColinEberhardt** [comment]
Thanks @siakaramalegos - can you please give me the required privileges to allow me to exit this issue?

**Comment by tunetheweb** [comment]
Invited you to the GitHub team there @ColinEberhardt

**Comment by ColinEberhardt** [comment]
@siakaramalegos I've completed a very rough outline, discussing with our analyst and reviewer. if you need any more detail, let me know.

We will likely be wanting to make some changes / additions to the crawler. I've started the process via a very simple tidy-up PR:

https://github.com/HTTPArchive/wasm-stats/pull/1

Once this is merged we're going to add a few new features. Should all be ready ahead of the June crawl.

**Comment by siakaramalegos** [comment]
@binji, @RReverser just a reminder that the outline is due in 2 days. Please request edit access to the google doc, add your name and email to confirm participation, and provide feedback on the outline that Colin started. This deadline is critical as we only have 2 more weeks afterward to finalize custom metrics. Thanks!

**Comment by siakaramalegos** [comment]
@ColinEberhardt and team, is the outline complete? The deadline was May 15. We now have less than 2 weeks to finalize any custom metrics so getting to outline completion is critical. When it's complete, please check the milestone off. Thanks for your contributions!

**Comment by ColinEberhardt** [comment]
I’m happy to call it complete - I discussed it verbally with @binji a few days ago. I’m busy working on custom metrics with @JamieWhitMac, so all progressing nicely

**Comment by tunetheweb** [comment]
In case you missed it: https://blog.scottlogic.com/2022/06/20/state-of-wasm-2022.html

**Comment by ColinEberhardt** [comment]
Hi @tunetheweb - yes, I did spot that. I'm the author 😀

**Comment by tunetheweb** [comment]
D'Oh! Missed that 🤣

**Comment by siakaramalegos** [comment]
@JamieWhitMac  it looks like the analysis is almost complete - can you give us an idea of % complete and timeline on the rest?

@ColinEberhardt  when do you think you can begin the draft? Just as a reminder, the due date at the end of the month is for post-review and post-edit, so you'll need to set aside at least a week for those and preferably more.

**Comment by ColinEberhardt** [comment]
Hi @siakaramalegos - my plan is to tackle the writing this week. I'll keep you updated on progress.

**Comment by RReverser** [comment]
Just so that you don't worry if I don't respond, I'll be away starting tomorrow till 24th of August. I'll be still happy to review the draft upon return.

**Comment by ColinEberhardt** [comment]
Thanks for the heads-up @RReverser  - I'll try to get something ready for you to review ahead of that.

**Comment by JamieWhitMac** [comment]
@siakaramalegos 95% complete, final 5% sorted this week (mostly just finalising visualisations and confirming we definitely have everything required).

**Comment by ColinEberhardt** [comment]
Hi @siakaramalegos the draft is complete:

https://docs.google.com/document/d/1U8CeQ_2ga_m-TGDAIGgB2PjB0O6Z5dauqtvuZubmXtQ/edit#heading=h.opp513127kig

@RReverser @binji @JamieWhitMac please do take a look and let me know if you have any comments.

**Comment by siakaramalegos** [comment]
@ColinEberhardt awesome! It looks like there are some comments on the queries PR so some data might need to be adjusted. Keep in mind for the draft. Might want to wait before reviewing.

**Comment by ColinEberhardt** [comment]
Hi @siakaramalegos thanks - can you link to the queries PR?

**Comment by siakaramalegos** [comment]
It's linked here already but for your convenience #2951

**Comment by siakaramalegos** [comment]
@ColinEberhardt Looks like the data in the sheet was updated so the analysis should be good. I did not cross reference the sheet data vs your draft so you might want to double-check if you haven't already.

@RReverser @binji Can you review Colin's draft within the next 1-2 days? After that, Colin will need time for any author edits, then we have another editing (writing style/usage) cycle to do all before Aug 31 if we still want to launch on time. Thanks for your help.

**Comment by RReverser** [comment]
Thanks for the ping. I skimmed through the draft, and overall it looks good to me. I might think of something later, but right now LGTM :)

**Comment by binji** [comment]
I read through it as well, wanted to leave a comment or two, but didn't have access? Anyway, nothing too important, mostly just a few suggestions/clarifications. Specifically I was thinking that some of the graphs might be nice to have comparisons with last year, to show the change. I think there may be one graph that's mislabeled as 2021 as well? And there is a reference to "WebIDL" that should be "WebIDL Bindings" I think. But other than that, it looks great!

**Comment by tunetheweb** [comment]
You can ask for access to make comments.

I corrected that 2021 graph label, and also resized the ones that were too big, and replaced them with links in the docs so they are linked to the sheet.

I also added the markup needed for the figures (was going to give you one example, but there one were a few so did them all).

I did notice while copying and pasting across that usage seems lower than last year. So do think that's worth commenting on (any theories from anyone?). I also copied across last years usage stats and created a year on year stat for desktop here, in case you want to use it: https://docs.google.com/spreadsheets/d/11jyABro0fKtuN6INnYP9pJlv5QWwp0jfJyTsGfKgScg/edit#gid=2142789475

**Comment by RReverser** [comment]
> So do think that's worth commenting on

That seems covered in the draft (with some theories, anyway). Given that HTTP Archive scanning is inconsistent and doesn't always include the same websites, it's not very surprising - I remember seeing month-to-month differences too, though. I wonder though, did the change where HTTP Archive is now traversing websites rather than only retrieving the homepage go live yet / did it affect the data?

**Comment by tunetheweb** [comment]
> That seems covered in the draft (with some theories, anyway).

Yeah I commented on some of the wording there.

> I wonder though, did the change where HTTP Archive is now traversing websites rather than only retrieving the homepage go live yet / did it affect the data?

That did go live, and we now scan one extra page per origin (the LCP link from the home page). It is in another dataset (the `2022_06_09` - ignore the data, it's the same run as the general June one) which includes both home pages and the secondary page. However, given it was JUST introduced the month of the Almanac run, we didn't overly encourage its use this year. But this chapter should feel free to have a look there if interested and it won't delay the chapter.

**Comment by RReverser** [comment]
Ah, if it's only one secondary page, it's probably unlikely to make much difference. I thought we're now traversing all pages directly reachable from main.

**Comment by tunetheweb** [comment]
That would be a LOT of pages! We're not Google Search scale just yet 😉

**Comment by RReverser** [comment]
One can dream!

**Comment by siakaramalegos** [comment]
@ColinEberhardt great job so far! Are the reviews all complete and incorporated? We're trying to determine whether it's ready for editing and the next step.

**Comment by ColinEberhardt** [comment]
@siakaramalegos yes, just a couple of comments which I've replied to which I don't believe require any further action.

It is good to go 🚢

**Comment by siakaramalegos** [comment]
@rviscomi in case you didn't already notice, this chapter is ready for editorial review

**Comment by siakaramalegos** [comment]
@rviscomi are you still editing this one or should I put it in the edit issue?

**Comment by rviscomi** [comment]
@siakaramalegos let's add this to the backlog in #3123. Thanks.
